hydra/_internal/hydra.py

```diff
@@ -103,15 +104,23 @@ def run(
             run_mode=RunMode.RUN,
         )
         HydraConfig.instance().set_config(cfg)
+        callbacks = get_callbacks(cfg)
 
-        return run_job(
+        for c in callbacks:
+            c.pre_run(cfg)
```

Comment:
an exception in a callback would prevent the execution of other callbacks.
While it's okay to allow it to flow to let the user know, we should execute all callbacks first.